go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -267,3 +268,115 @@ func (app *rootHashApplication) executorCommit(
 
 	return nil
 }
+
+func (app *rootHashApplication) submitEvidence(
+	ctx *abciAPI.Context,
+	state *roothashState.MutableState,
+	evidence *roothash.Evidence,
+) error {
+	// Validate proposal content basics.
+	if err := evidence.ValidateBasic(); err != nil {
+		ctx.Logger().Error("Evidence: submitted evidence not valid",
+			"evidence", evidence,
+			"err", err,
+		)
+		return fmt.Errorf("%w: %v", roothash.ErrInvalidEvidence, err)
+	}
+
+	if ctx.IsCheckOnly() {
+		return nil
+	}
+
+	// Charge gas for this transaction.
+	params, err := state.ConsensusParameters(ctx)
+	if err != nil {
+		ctx.Logger().Error("Evidence: failed to fetch consensus parameters",
+			"err", err,
+		)
+		return err
+	}
+	if err = ctx.Gas().UseGas(1, roothash.GasOpEvidence, params.GasCosts); err != nil {
+		return err
+	}
+
+	rtState, _, _, err := app.getRuntimeState(ctx, state, evidence.ID)
+	if err != nil {
+		return err
+	}
+
+	if len(rtState.Runtime.Staking.Slashing) == 0 {
```

Comment:
But it could be nil and derefing a nil map is an error. In case of `Slashing` from the (global) staking consensus parameters, the `Slashing()` state accessor method returns a non-nil map in case the parameters are unset/nil but here this is not the case.

Reply:
https://golang.org/ref/spec#Index_expressions language spec says reading from a nil map gives the zero value